orttraining/orttraining/core/graph/training_op_defs.cc

```diff
@@ -1487,6 +1485,118 @@ Example 4:
+      .TypeConstraint(
+          "T",
+          {"tensor(bool)", "tensor(float)"},
+          "boolean or float uncompressed tensors.")
+      .TypeConstraint(
+          "T1",
+          {"tensor(uint8)"},
+          "8 bits represent 8 1-bit compressed tensors.");
+
+  ONNX_CONTRIB_OPERATOR_SCHEMA(GistPack1Decoder)
+      .SetDomain(kOnnxDomain)
+      .SinceVersion(9)
+      .Input(0, "X", "1 bit compresssed input", "T1")
+      .Output(0, "Y", "uncompressed output", "T")
+      .TypeConstraint(
+          "T1",
+          {"tensor(uint8)"},
+          "8 bits represent 8 1-bit compressed tensors.")
+      .TypeConstraint(
+         "T",
+          {"tensor(bool)", "tensor(float)"},
+          "boolean or float uncompressed tensors.");
+
+  ONNX_CONTRIB_OPERATOR_SCHEMA(GistPack8Encoder)
+      .SetDomain(kOnnxDomain)
+      .SinceVersion(9)
+      .Input(0, "X", "uncompressed input", "T")
+      .Output(0, "Y", "compressed output", "T1")
+      .TypeConstraint(
+          "T",
+          {"tensor(float16)", "tensor(float)"},
+          "Constrain to all numeric tensors.")
+      .TypeConstraint(
+          "T1",
+          {"tensor(uint8)"},
+          "8 bits compressed tensors.");
+  
+  ONNX_CONTRIB_OPERATOR_SCHEMA(GistPack8Decoder)
+      .SetDomain(kOnnxDomain)
+      .SinceVersion(9)
+       .Input(0, "X", "compresssed input", "T1")
+       .Output(0, "Y", "uncompressed output", "T")
+      .TypeConstraint(
+          "T",
+          {"tensor(float16)", "tensor(float)"},
+          "Constrain to all numeric tensors.")
+      .TypeConstraint(
+          "T1",
+          {"tensor(uint8)"},
+          "8 bits compressed tensors.");
+
+  ONNX_CONTRIB_OPERATOR_SCHEMA(GistPack16Encoder)
+      .SetDomain(kOnnxDomain)
+      .SinceVersion(9)
+      .Input(0, "X", "uncompressed input", "T")
+      .Output(0, "Y", "compressed output", "T1")
+      .TypeConstraint(
+          "T",
+          {"tensor(float)"},
+          "Constrain to all numeric tensors.")
+      .TypeConstraint(
+          "T1",
+          {"tensor(float16)"},
+          "16 bits compressed tensors.");
+
+  ONNX_CONTRIB_OPERATOR_SCHEMA(GistPack16Decoder)
+      .SetDomain(kOnnxDomain)
+      .SinceVersion(9)
+      .Input(0, "X", "compressed input", "T1")
+      .Output(0, "Y", "uncompressed output", "T")
+      .TypeConstraint(
+          "T",
+          {"tensor(float)"},
+          "Constrain to all numeric tensors.")
+      .TypeConstraint(
+          "T1",
+          {"tensor(float16)"},
+          "16 bits compressed tensors.");
+
+  ONNX_CONTRIB_OPERATOR_SCHEMA(GistPackMsfp15Encoder)
+      .SetDomain(kOnnxDomain)
+      .SinceVersion(9)
+      .Input(0, "X", "uncompressed input", "T")
+      .Output(0, "Y", "compressed output", "T1")
+      .TypeConstraint(
+          "T",
+          {"tensor(float)"},
+          "Constrain to all numeric tensors.")
+      .TypeConstraint(
+          "T1",
+          {"tensor(uint8)"},
+          "8 bits compressed tensors.");
```

Comment:
Not sure if I understand the question completely. 
For Msfp15 we compress a tile size (16 elements) worth of float data to tile size worth of uint8 data. This 16 * uint8 data consists of the 16 * 7-bit (sign + mantissa) as well as a 8-bit exponent. Based on this, the "8 bits compressed tensors" is correct for the case of encoder and decoder for GistPackMsfp15.

Reply:
A bit of a pattern match: I think my confusion here came from the "8 bits" here not matching the "Msfp15" in the op name, whereas in the other ops it does match, or say something more descriptive (e.g. or GistPack1Encoder, "8 bits represent 8 1-bit compressed tensors"). Based on your explanation, I still think "8 bits compressed" is inaccurate  because it says nothing about the tiling in MSFP - just the packaging format. 

For example, if you used `tensor(uint32)` with 4 values per tile, would it be as meaningful to say "32 bits compressed tensors"?

By contrast, for GistPack8Encoder/Decoder, you also say "8 bits compressed tensors", which (IIUC) means something different. 

Could you rewrite it to say "8 bits represents <X>" for some explanation <X>?

Reply:
Updated the explanation of how compressed tensor in Msfp15 case encode the data (sign + mantissa, and shared exponent)